docs/setup_installation/azure/getting_started.md

```diff
@@ -24,78 +24,137 @@ To run all the commands on this page the user needs to have at least the followi
+  "Actions": [
+    "Microsoft.Storage/storageAccounts/blobServices/containers/write",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/read",
+    "Microsoft.Storage/storageAccounts/blobServices/write",
+    "Microsoft.Storage/storageAccounts/blobServices/read",
+    "Microsoft.Storage/storageAccounts/listKeys/action"
+  ],
+  "NotActions": [],
+  "DataActions": [
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/delete",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/read",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/move/action",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write"
+  ],
+  "AssignableScopes": [
+    "'$STORAGE_ID'"
+  ]
+}'
+
+sleep 30  # give Azure some time to persist the new role
+
+az role assignment create --role hopsfs-storage-permissions --assignee $UA_IDENTITY_PRINCIPAL_ID --scope $STORAGE_ID
+```
 
-export storage_id=`az storage account show --name $storage_account_name --resource-group $resource_group --query "id" --output tsv`
+Also, assign roles to pull/push images from/to the container registry:
 
-export acr_id=`az acr show --name $registry_name --resource-group $resource_group --query "id" --output tsv`
+```bash
+az role assignment create --assignee $UA_IDENTITY_PRINCIPAL_ID --role AcrPull --scope $ACR_ID
+az role assignment create --assignee $UA_IDENTITY_PRINCIPAL_ID --role "AcrPush" --scope $ACR_ID
+az role assignment create --assignee $UA_IDENTITY_PRINCIPAL_ID --role "AcrDelete" --scope $ACR_ID
 ```
 
-### Step 1.5: Assign Roles to Managed Identity
+### Step 1.5: Create Service Principal for Hopsworks services
 
-```bash
-az role assignment create --assignee $managed_id --role "Storage Blob Data Contributor" --scope $storage_id
+Create a service principal to grant Hopsworks applications with access to the container registry. For example, Hopsworks uses this service principal to push new Python environments created via the Hopsworks UI.
 
-az role assignment create --assignee $managed_id --role AcrPull --scope $acr_id
-az role assignment create --assignee $managed_id --role "AcrPush" --scope $acr_id
-az role assignment create --assignee $managed_id --role "AcrDelete" --scope $acr_id
+```bash
+export SP_PASSWORD=`az ad sp create-for-rbac --name $SP_NAME --scopes $ACR_ID --role acrpush --query "password" --output tsv`
+export SP_USER_NAME=`az ad sp list --display-name $SP_NAME --query "[].appId" --output tsv`
 ```
 
-### Step 1.6: Allow AKS cluster access to ACR repository.
+### Step 1.6: Create an AKS Kubernetes Cluster
+
+Provision an AKS cluster with a number of nodes:
 
 ```bash
-az aks update --resource-group $resource_group --name $cluster_name --attach-acr $registry_name
+az aks create --resource-group $RESOURCE_GROUP --name $KUBERNETES_CLUSTER_NAME --network-plugin azure \
+    --enable-cluster-autoscaler --min-count 1 --max-count 4 --node-count 3 --node-vm-size Standard_D16_v4 \
```

Comment:
these are 16 cpus, 64G nodes, do we need that many nodes? should we have the 32G instead?

Reply:
Right, that's 64G. It's the instance type we had before. I'll change it to `Standard_D8_v4` (32G, 8 cores)